### PR TITLE
Added form to new user handling in sign up

### PR DESCRIPTION
**Pull request by markwhiting**
Added a link to google form when `register.error && form.$pristine` are true for a users submission.

### CODE REVIEW

File: static/templates/authentication/register.html [hunk 1]
```diff
@@ -74,8 +74,9 @@
         you agree to provide Daemo accurate and complete information about you and your business, and you authorize Daemo to share it and transaction
         information related to your use of the payment processing services provided by Stripe. Daemo does not store
         any banking or credit card information.</div-->
-    <div class="authentication-error" layout="row" ng-show="register.error && form.$pristine"
-         ng-bind="register.error"></div>
+    <div class="authentication-error" layout="row" ng-show="register.error && form.$pristine">
+      <span>We are currently in closed beta. Please <a href="https://goo.gl/forms/8eNjaR8pKc3aIpci1">fill out this form</a> to register interest in using Daemo.</span>
```
Comment (dmorina) [inline static/templates/authentication/register.html]: register error can be anything here, can you check in angular if the response code is 403 or whatever you get when registration closed exception is raised or worst case you can do a string comparison of current error message

**Comment by markwhiting** [comment]
@dmorina, I updated to include that check.

File: static/templates/authentication/register.html [hunk 2]
```diff
@@ -74,8 +74,9 @@
         you agree to provide Daemo accurate and complete information about you and your business, and you authorize Daemo to share it and transaction
         information related to your use of the payment processing services provided by Stripe. Daemo does not store
         any banking or credit card information.</div-->
-    <div class="authentication-error" layout="row" ng-show="register.error && form.$pristine"
-         ng-bind="register.error"></div>
+    <div class="authentication-error" layout="row" ng-show="register.error=='We are currently in closed beta. If you\'d like an account, email daemo@cs.stanford.edu with a short description of what you\'d like to use Daemo for.' && form.$pristine">
```
Comment (dmorina) [inline static/templates/authentication/register.html]: and what if it's something else? :P 
i'd recommend doing this in the controller

Comment (shirishgoyal) [inline static/templates/authentication/register.html]: I think better to hide the signup button or even the form itself till registration is not open. Also the link should open on new page (target="_blank"). Can we change the text to something like -- fill the form to add yourself to the waitlist and we will reach out to you as soon as we have more slots open for you.

**Comment by mbernst** [comment]
Don't we need the form open in order to enroll new requesters?

**Comment by markwhiting** [comment]
I’d previously thought of pushing the sign up form to a secret link and just making the data collection form link directly from the ‘getting started’ link. 

This might be more tricky, but might also make things easier for users in the short term.

**Comment by markwhiting** [comment]
@dmorina updated with those changes.

File: static/templates/authentication/register.html [hunk 3]
```diff
@@ -74,8 +74,9 @@
         you agree to provide Daemo accurate and complete information about you and your business, and you authorize Daemo to share it and transaction
         information related to your use of the payment processing services provided by Stripe. Daemo does not store
         any banking or credit card information.</div-->
-    <div class="authentication-error" layout="row" ng-show="register.error && form.$pristine"
```
Comment (dmorina) [inline static/templates/authentication/register.html]: Keep this around in case it's not a 403

**Comment by markwhiting** [comment]
@dmorina, Ok, sure. I've added this and filtered out when it is a 403, to avoid 2 errors appearing.

Not sure if thats the right way to go about it, let me know what you think.

File: static/js/authentication/controllers/register.controller.js [hunk 4]
```diff
@@ -41,6 +42,11 @@
                             $mdToast.showSimple('Email with an activation link has been sent.');
                             $state.go('auth.login');
                         }, function (data, status) {
+                            
+                            if (status === 403) {
```
Comment (dmorina) [inline static/js/authentication/controllers/register.controller.js]: alright, status seems to be undefined, don't use that then just do `data.status`

File: static/templates/authentication/register.html [hunk 5]
```diff
@@ -74,8 +74,10 @@
...
-         ng-bind="register.error"></div>
+    <div class="authentication-error" layout="row" ng-show="registrationEnabled == false && form.$pristine">
```
Comment (dmorina) [inline static/templates/authentication/register.html]: registrationEnabled isn't in the scope and are you setting the form to $pristine if you are using it as a condition here?

**Pull request merged**